README.md

```diff
@@ -1 +1,588 @@
+    return new Promise((resolve) => {
+      this.__resolve = resolve;
+    });
+  }
+
+  set a(val) {
+    this.__a = val;
+    this.requestUpdate();
+  }
+
+  set b(val) {
+    this.__b = val;
+    this.requestUpdate();
+  }
+}
+
+/* We use an IIFE (Immediately Invoked Function Expression), because top-level await is not a thing yet 😑 */
+(async () => {
+  /* We instantiate a new instance of our class */
+  const batching = new Batching();
+
+  /* Set multiple properties in a row */
+  batching.a = 1;
+  batching.b = 2;
+
+  /* And this is where we `await` an update */
+  await batching.updateComplete;
+
+  /* We then assign another property */
+  batching.b = 3;
+
+  /** 
+   * 🎉 The result:
+   * "updating!" is logged to the console twice!
+   */
+})();
+```
+
+## Back to web components land
+
+Sweet, now that we have a solid understanding of using microtasks to our advantage to batch work, let's go back to web components land, and see how we could implement this as a simple base class for web components.
+
+```js
+/* We create a new class that extends from the native HTMLElement */
+class BatchingElement extends HTMLElement {
+  constructor() {
+
+    /* We now have to call `super` to make sure our element gets set up correctly */
+    super();
+    this.updateComplete = this.__createDeferredPromise();
+  }
+
+  async requestUpdate() {
+    if (!this.updateRequested) {
+      this.updateRequested = true;
+      this.updateRequested = await false;
+      this.update();
+      this.__resolve();
+      this.updateComplete = this.__createDeferredPromise();
+    }
+  }
+
+  update() {}
+
+  __createDeferredPromise() {
+    return new Promise((resolve) => {
+      this.__resolve = resolve;
+    });
+  }
+}
+```
+
+> Again, this code is a simplified example of the technique LitElement uses for illustration purposes, if you're interested in seeing how LitElement achieves this exactly, you can find the source code [here](https://github.com/Polymer/lit-element/blob/master/src/lib/updating-element.ts#L715).
+
+And, well, that's it. Not much has changed here really, other than the fact that we now extend from `HTMLElement`, and have to call `super()` in the constructor. Let's see how we can actually use it in a component:
+
+```js
+/* We create a new class, MyElement, that extends from our BatchingElement base class */
+class MyElement extends BatchingElement {
+
+  constructor() {
+    super();
+
+    /* We attach a shadowRoot to our component for good measure */
+    this.attachShadow({ mode: "open" });
+  }
+
+  connectedCallback() {
+    /* And we call an initial render when our component gets connected to the DOM */
+    this.update();
+  }
+
+  /**
+   * This is our `update` function that will get triggered by the
+   * `requestUpdate` method.
+   * 
+   * Any time we set a property on this element, we'll also
+   * trigger an update.
+   */
+  update() {
+    console.log('updating!');
+    this.shadowRoot.innerHTML = `
```

Comment:
yeah seeing this again gives me twitter flashbacks. I say never put `shadowRoot.innerHTML = ` in a public post even if it' explicitly a naive implemention